internal/internal_workflow_testsuite.go

```diff
@@ -2431,6 +2431,10 @@ func (env *testWorkflowEnvironmentImpl) setStartWorkflowOptions(options StartWor
 		wf.WorkflowTaskTimeout = options.WorkflowTaskTimeout
 	}
 	if len(options.ID) > 0 {
+		// Reassign the ID in running Workflows so SignalWorkflowByID can find the workflow
+		originalID := wf.WorkflowExecution.ID
+		env.runningWorkflows[options.ID] = env.runningWorkflows[wf.WorkflowExecution.ID]
+		delete(env.runningWorkflows, originalID)
```

Comment:
I was originally thinking maybe we wanted to also leave it in the map under the original ID, but if someone is referring to it by original ID in their tests even after setting w/ a new ID, I think they can be asked to fix their tests. Thoughts?

Reply:
yeah  I think it is a reasonable to  ask they fix their test